packages/lib-utils/src/components/status/StatusBox.tsx

```diff
@@ -0,0 +1,99 @@
+import { Alert, EmptyState, EmptyStateBody, Spinner, Text, Title } from '@patternfly/react-core';
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+export type LoadError = {
+  message: string;
+  status: number;
+};
+
+type AccessDeniedProps = {
+  message: string | React.ComponentType;
+};
+
+export const AccessDenied: React.FC<AccessDeniedProps> = ({
+  message = 'Insufficient access rights.',
+}) => (
+  <EmptyState>
+    <Title headingLevel="h2">Restricted Access</Title>
+    <EmptyStateBody>
+      <Text>You don&apos;t have access to this section due to cluster policy.</Text>
+      {message && (
+        <Alert isInline variant="danger" title="Error details">
+          {message}
+        </Alert>
+      )}
+    </EmptyStateBody>
+  </EmptyState>
+);
+
+export type StatusBoxProps<D> = {
+  areFiltersApplied?: boolean;
+  data?: D[];
+  children?: React.ReactElement;
+  emptyLabel?: string;
+  loadError?: LoadError;
+  loaded?: boolean;
+  LoadErrorDefaultMsg?: React.ComponentType<Record<string, never>>;
+  NoDataEmptyMsg?: React.ComponentType<Record<string, never>>;
+  EmptyMsg?: React.ComponentType<Record<string, never>>;
+};
+
+export const StatusBox: React.FC<StatusBoxProps<Record<string, unknown>>> = ({
```

Comment:
You can use the existing `AnyObject` type which is aliased to `Record<string, unknown>`.

```ts
import type { AnyObject } from '@monorepo/common';
```

I'd suggest doing a search & replace where
- `Record<string, unknown>` :arrow_right: `AnyObject`
- `Record<string, never>` :arrow_right: `EmptyObject`